sdk-setup/src/mb2

```diff
@@ -2326,58 +2326,57 @@ run_deploy() {
     [[ ! $all_opt || $method_opt != --rsync ]] || fatal "Cannot use '--all' with '--rsync'"
     [[ ${method_opt:-} ]] || fatal "Argument expected"
 
-    local retcode=0
+    # rsync method is special
+    if [[ $method_opt == --rsync ]]; then
+        ! behind_qtcreator || run_make_install || return
+        local name=$(get_spec_tag "%{name}")
+        rsync_as $device_username -av ${buildroot}/. $device_ip:/opt/sdk/$name
+        return
+    fi
+
+    if behind_qtcreator; then
+        run_package || return
+    fi
+
+    local rpms=$(find_rpms $all_opt "$OPT_OUTPUTDIR" "$STATEDIR/spec" "${patterns[@]}")
+    [[ $rpms ]] || fatal "No package found or none matched the given criteria"
+
+    local remote_username=$device_username
+    local method=()
+    local ok_retcodes=(0)
 
     case $method_opt in
         --sdk | --pkcon )
-            local method;
             if [[ $method_opt == --sdk ]]; then
-                method="sdk-deploy-rpm"
+                method=(sdk-deploy-rpm)
             else
-                method="pkcon --plain --noninteractive install-local"
+                method=(pkcon --plain --noninteractive install-local)
```

Comment:
Now this is just silly. Why don't you have separate cases for --sdk and --pkcon and get rid of the if?

Reply:
Simply because I was just moving code around and facing other issues I didn't spot this one :)